pgscm/admin/views.py

```diff
@@ -39,6 +46,80 @@ def users():
                 (p.province_id, p.type + " " + p.name) for p in
                 models.Province.query.order_by(
                     models.Province.name.asc()).all()]
+        form.roles.choices = [
+            (r.id, r.description) for r in
+            models.Role.query.order_by(
+                models.Role.name.asc()).all()]
+
+        # form create or edit submit
+        if request.method == 'POST' and form.data['submit']:
+            if not check_role(crud_role):
+                return redirect(url_for(request.endpoint))
+            elif form.validate_on_submit():
+                # edit user
+                if form.id.data:
+                    edit_user = sqla.session.query(models.User) \
+                        .filter_by(id=form.id.data).one()
+                    if edit_user.email != form.email.data and \
+                            email_is_exist(form.email.data):
+                        flash(str(__('Email existed!')), 'error')
```

Comment:
for creating, checking and set user role, see example here:
1. Checking and creating new user: https://github.com/hieulq/pgscm/blob/master/pgscm/__init__.py#L133
2. Add role to new user: https://github.com/hieulq/pgscm/blob/master/pgscm/__init__.py#L158

No need to create methods for checking existed email or something else, we use flask_security for handling user and role management, please read flask_security docs.